opm/simulators/flow/FlowMainEbos.cpp

```diff
@@ -77,6 +77,84 @@ void handleExtraConvergenceOutput(SimulatorReport& report,
       report.fullReports(os);
     }
 }
+void checkAllMPIProcesses()
+{
+#if HAVE_MPI
+    const auto& comm = EclGenericVanguard::comm();
+    if (comm.size() > 1)
+    {
+        // we try to prevent the abort here.
+        // For that we need a signal that each process is here.
+        // Each process sends  a message to rank 0.
+        const int tag = 357912;
+        if (comm.rank() == 0)
+        {
+            // wait for a message from all processes.
+            std::vector<MPI_Request> requests(comm.size() - 1, MPI_REQUEST_NULL);
+            std::vector<int> data(comm.size()-1);
+
+            for(decltype(comm.size()) i = 1; i < comm.size(); ++i)
+            {
+                if (auto error = MPI_Irecv(data.data() + i, 1, MPI_INT, i, tag, comm, requests.data() + i - 1);
```

Comment:
I believe there is a off-by-one bug here, if for example comm has size 2, then data will have size 1, but we will pass a pointer pointing to the (non-existing) element 1.

Reply:
Indeed we write out of bounds here.